components/skus/browser/rs/lib/src/sdk/credentials/fetch.rs

```diff
@@ -67,6 +79,127 @@ where
+                                            .collect();
+
+                                    let blinded_creds: Vec<BlindedToken> =
+                                        creds.iter().map(|t| t.blind()).collect();
+
+                                    if !initialized {
+                                        self.client.upsert_time_limited_v2_item_creds(&item.id, creds).await?;
+                                    }
+                                    blinded_creds
+                                } else {
+                                    // not almost expired, already initialized
+                                    Vec::<BlindedToken>::new() // nothing to do
+                                }
+                            }
+                            None => {
+                                // we have no idea about this order item
+                                let creds: Vec<Token> =
+                                    iter::repeat_with(|| Token::random::<Sha512, _>(&mut csprng))
+                                        .take(num_creds)
+                                        .collect();
+
+                                let blinded_creds: Vec<BlindedToken> =
+                                    creds.iter().map(|t| t.blind()).collect();
+
+                                self.client.upsert_time_limited_v2_item_creds(&item.id, creds).await?;
+                                blinded_creds
+                            }
+                        };
+
+                    // if the blinded_creds length is less than 1 we do not want to submit this
+                    // request.  this happens if we currently are full with credentials
+
+                    if blinded_creds.len() > 0 {
+                        let claim_req = ItemCredentialsRequest { item_id: item.id, blinded_creds };
+
+                        let request_with_retries = FutureRetry::new(
+                            || async {
+                                let body = serde_json::to_vec(&claim_req)
+                                    .or(Err(InternalError::SerializationFailed))?;
+
+                                let req = http::Request::builder()
+                                    .method("POST")
+                                    .uri(format!(
+                                        "{}/v1/orders/{}/credentials",
+                                        self.base_url, order_id
+                                    ))
+                                    .body(body)
+                                    .unwrap();
+
+                                let resp = self.fetch(req).await?;
+
+                                match resp.status() {
+                                    http::StatusCode::OK => Ok(()),
+                                    http::StatusCode::NOT_FOUND => Err(InternalError::NotFound),
+                                    _ => Err(resp.into()),
+                                }
```

Comment:
Can `resp.into()` handle the `NOT_FOUND` -> `InternalError::NotFound` case?

Reply:
https://github.com/brave/brave-core/pull/15465/commits/9c9188c99dcc3e8bb08fd4835f602a62155975ae